api_spec.yaml

```diff
@@ -3246,9 +3246,451 @@ paths:
+                  type: boolean
+                  example: false
+
+      responses:
+        200:
+          description: Create or Update result
+          content:
+            application/json:
+              schema:
+                oneOf:
+                - type: object
+                  required:
+                    - status
+                    - createdNew
+                  properties:
+                    status:
+                      $ref: '#/components/schemas/statusOK'
+                    createdNew:
+                      type: boolean
+                - type: object
+                  required:
+                  - status
+                  - reason
+                  properties:
+                    status:
+                      type: string
+                      enum: ["CONFIG_VALIDATION_ERROR"]
+                    reason:
+                      type: string
+                - type: object
+                  required:
+                    - status
+                  properties:
+                    status:
+                      $ref: '#/components/schemas/statusTenantDoesNotExistError'
+        '400':
+          $ref: '#/components/responses/400'
+        
+        '401':
+          $ref: '#/components/responses/401'
+        
+        '402':
+          $ref: '#/components/responses/402'
+
+        '404':
+          $ref: '#/components/responses/404'
         
+        '500':
+          $ref: '#/components/responses/500'
+
+  /recipe/multitenancy/config/thirdparty/provider/remove:
+    post:
+      tags:
+        - Multitenancy Recipe
+      description: |
+        Delete ThirdParty Provider config for a tenant.
+
+        If `tenantId` is not provided, `defaultTenantId` will be used.
+      operationId: deleteTenantConfigPOST
+      parameters:
+        - $ref: '#/components/parameters/multitenancyRid'
+        - $ref: '#/components/parameters/api-key'
+        - $ref: '#/components/parameters/cdi-version'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - thirdPartyId
+              properties:
+                thirdPartyId:
+                  $ref: '#/components/schemas/thirdPartyId'
+                tenantId:
+                  $ref: '#/components/schemas/tenantId'
+
+      responses:
+        200:
+          description: Redirect URL
+          content:
+            application/json:
+              schema:
+                oneOf:
+                - type: object
+                  properties:
+                    status:
+                      $ref: '#/components/schemas/statusOK'
+                    didConfigExist:
+                      type: boolean
+                - type: object
+                  required:
+                    - status
+                  properties:
+                    status:
+                      $ref: '#/components/schemas/statusTenantDoesNotExistError'
```

Comment:
if you are returning `statusTenantDoesNotExistError`, why do you need `didConfigExist`? That being said, I think you should not have `statusTenantDoesNotExistError` status type in this API at all.

Reply:
On second thought, you should readd `statusTenantDoesNotExistError ` to this APIs output. I had earlier misunderstood it as thirdPartIdDoesNotExistError

Reply:
Done